### ISSUE TITLE
 cli.isMultipleCompiler is not a function

### ISSUE THREAD

**Issue by 779102260** [webpack/webpack-cli#3294]
**Describe the bug**

@webpack-cli/serve 1.7.0： cli.isMultipleCompiler is not a function

isMultipleCompiler is add in 1.7.0

![image](https://user-images.githubusercontent.com/16102900/173278504-30c84149-2e3f-4d4f-9503-a293928b44d0.png)

code in v1.6.1 

```js
const compilers = typeof compiler.compilers !== "undefined" ? compiler.compilers : [compiler];
```
should be `cli.isMultipleCompiler?.(compiler)` ?

**Comment by snitin315** [comment]
Are you using webpack-cli v4.10.0?

**Comment by 779102260** [comment]
> 
no, using 4.9.0，so i need to update to 4.10.0 ?
![image](https://user-images.githubusercontent.com/16102900/173281734-a22ec9ed-46bc-4043-9f1e-fe7770cd603c.png)

**Comment by snitin315** [comment]
@779102260 yes, please update webpack-cli and related packages to the latest version.

**Comment by 779102260** [comment]
> @779102260 yes, please update webpack-cli and related packages to the latest version.

it work, thx

**Comment by snitin315** [comment]
@alexander-akait should we keep it compatible with older @webpack-cli/serve v1.6?

**Comment by 779102260** [comment]
yes, if not too much work to do



---Original---
From: "Nitin ***@***.***&gt;
Date: Mon, Jun 13, 2022 16:53 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [webpack/webpack-cli]  cli.isMultipleCompiler is not a function(Issue #3294)




 
@alexander-akait should we keep it compatible with older @webpack-cli/serve v1.6?

**Comment by simasch** [comment]
Please don't use version ranges in transitive dependencies!

**Comment by TatuLund** [comment]
> Please don't use version ranges in transitive dependencies!

I agree. Nowadays supply chain safety and quality is a major concern. Organizations who use these tools professionally need to apply quality gates in CI and processes. Hence they need to use pinned versions instead open ended versions. I.e. they can't allow maven or npm to load what ever newer version automatically. The new versions need to be tested and verified for compatibility and safety. Hence I think all essential tools related to building apps should use exact versions also for transitive dependencies.
